app/views/admin/admins/index.html.haml

```diff
@@ -1,21 +1,5 @@
-%h1 Admins Management
+%h1 Admin Management
 
 %p= link_to "Add new admin", new_admin_admin_path, class: "btn"
 
-%table
-  %col{style: 'width: 50%'}
-  %thead
-    %tr
-      %th= sortable_heading_for(:email)
-      %th.actions Actions
-  %tbody
-    - @users.each do |user|
-      %tr
-        %td= user.email
-        %td.actions
-          - if policy(user).edit?
-            = link_to "Edit", edit_admin_admin_path(user), class: "btn btn-small"
-          - if policy(user).destroy?
-            = link_to "Delete", admin_admin_path(user), method: :delete, data: { confirm: "Are you sure you want to delete #{user}?" }, class: "btn btn-small"
-
-= paginate @users
+= render partial: "user_results", locals: {delete_url: -> (user) { admin_admin_path(user) }, edit_url: -> (user) { edit_admin_admin_path(user) } }
```

Comment:
I think you don't need to provide these locals, I'd do:

```ruby
= render partial: "user_results", locals: { users: @users }
```
 and in `_user_results.html.haml`:

```ruby
= simple_form_for :search, method: :get do |s|
  = s.input :search_term, label: "User email search", placeholder: "type email here", input_html: { value: (params[:search][:search_term] unless params[:search].blank?) }
  = s.button :submit, "Search"

- if users.present?
  %table
    %colgroup
      %col{style: 'width: 80%'}
      %col{style: 'width: 20%'}
    %thead
      %tr
        %th= sortable_heading_for(:email)
        %th.actions Actions
    %tbody
      - users.each do |user|
        %tr
          %td= user.email
          %td.actions
            - if policy(user).edit?
              = link_to "Edit", edit_admin_admin_path(user), class: "btn btn-small"
            - if policy(user).destroy?
              = link_to "Delete", admin_admin_path(user), method: :delete, data: { confirm: "Are you sure you want to delete #{user}?" }, class: "btn btn-small"
  = paginate users

- else
  %p No users matched that search.
```